base/error.go

```diff
@@ -217,6 +217,14 @@ func IsDocNotFoundError(err error) bool {
 	}
 }
 
+// IsHTTPErrorStatusNotFound returns true if the error is a HTTP Error with the status 404 Not Found
+func IsHTTPErrorStatusNotFound(err error) bool {
```

Comment:
It looks like we're only using this in the context of GetActiveDatabase returning not found.  I don't think it makes sense to format the error as an HTTP error, and then have to parse that error again to see if it represents 'not found'.  Can we just have GetActiveDatabase return a non-HTTP error like ErrNotFound, or have it return a flag for found/not found?  Our rest handlers call ErrorAsHTTPStatus when returning errors, so I don't think we need to prematurely convert to an HTTP error.